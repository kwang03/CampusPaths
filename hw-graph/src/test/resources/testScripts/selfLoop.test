#Tests functionality of self loops in a graph

#Create graph
CreateGraph myGraph

#Adds some nodes to the graph
AddNode myGraph n1
AddNode myGraph n2
AddNode myGraph n3
AddNode myGraph n4
AddNode myGraph n5

#Independent self loop node
AddEdge myGraph n1 n1 e11
ListChildren myGraph n1

#Self loop node with edge to another node
AddEdge myGraph n2 n2 e22
AddEdge myGraph n2 n3 e23
ListChildren myGraph n2

#Self loop with another incoming edge
AddEdge myGraph n4 n4 e44
AddEdge myGraph n5 n4 e54
ListChildren myGraph n4
ListChildren myGraph n5