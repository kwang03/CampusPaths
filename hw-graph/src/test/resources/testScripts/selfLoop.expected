#Tests functionality of self loops in a graph

#Create graph
created graph myGraph

#Adds some nodes to the graph
added node n1 to myGraph
added node n2 to myGraph
added node n3 to myGraph
added node n4 to myGraph
added node n5 to myGraph

#Independent self loop node
added edge e11 from n1 to n1 in myGraph
the children of n1 in myGraph are: n1(e11)

#Self loop node with edge to another node
added edge e22 from n2 to n2 in myGraph
added edge e23 from n2 to n3 in myGraph
the children of n2 in myGraph are: n2(e22) n3(e23)

#Self loop with another incoming edge
added edge e44 from n4 to n4 in myGraph
added edge e54 from n5 to n4 in myGraph
the children of n4 in myGraph are: n4(e44)
the children of n5 in myGraph are: n4(e54)